File: TheOffice/CSharp/TheOffice/Room.cs
using System;
using System.Collections.Generic;

namespace TheOffice
{
    public class Room
    {
        private static Random RandomInteger = new Random();

        public Room North { get; set; }
        public Room South { get; set; }
        public Room East { get; set; }
        public Room West { get; set; }
        public List<Worker> Workers { get; set; }

        public Room()
        {
            this.Workers = new List<Worker>();
        }

        public Worker CreateNewWorker(string name)
        {
            Worker worker = new Worker(name, this);
            this.Workers.Add(worker);
            return worker;
        }

        public Room GetRandomExit()
        {
            List<Room> availableExits = new List<Room>();
            if (this.North != null) availableExits.Add(this.North);
            if (this.South != null) availableExits.Add(this.South);
            if (this.East != null) availableExits.Add(this.East);
            if (this.West != null) availableExits.Add(this.West);
            return availableExits[RandomInteger.Next(availableExits.Count)];
        }
    }
}

File: The Office/VisualBasic/TheOffice/Room.vb
Public Class Room

    Public Property North As Room
    Public Property South As Room
    Public Property East As Room
    Public Property West As Room
    Public Property Workers As List(Of Worker)

    Public Sub New()
        Me.Workers = New List(Of Worker)
    End Sub

    Public Function CreateNewWorker(Name As String) As Worker
        Dim Worker As Worker = New Worker(Name, Me)
        Me.Workers.Add(Worker)
        Return Worker
    End Function

    Public Function GetRandomExit() As Room
        Dim AvailableExits As List(Of Room) = New List(Of Room)
        If Me.North IsNot Nothing Then availableExits.Add(Me.North)
        If Me.South IsNot Nothing Then availableExits.Add(Me.South)
        If Me.East IsNot Nothing Then availableExits.Add(Me.East)
        If Me.West IsNot Nothing Then availableExits.Add(Me.West)
        Return AvailableExits(CInt(Math.Ceiling(Rnd() * (AvailableExits.Count))) - 1)
    End Function

End Class